src/apps/competitions/consumers.py

```diff
@@ -87,12 +82,12 @@ async def receive(self, text_data=None, bytes_data=None):
         if submission_ids:
             # Filter out submissions not by this user
             submissions = Submission.objects.filter(id__in=submission_ids, owner=self.scope["user"])
+            con = get_redis_connection("default")
 
             for sub in submissions:
-                text_path = os.path.join(settings.TEMP_SUBMISSION_STORAGE, f"{sub.id}.txt")
-                if os.path.exists(text_path):
-                    with open(text_path) as f:
-                        text = f.read()
+                text = (con.get(f':1:submission-{sub.id}-log'))
+                if text:
+                    text = str(text, 'utf-8')
                     await self.group_send(text, sub.id, full_text=True)
```

Comment:
I think you can inline do `text.encode('utf-8')` or whatever?

```py
await self.group_send(text.encode('utf-8'), sub.id, full_text=True)
```

Reply:
If text returns `None` then that will throw an error.